applications/StructuralMechanicsApplication/custom_response_functions/response_utilities/adjoint_nodal_reaction_response_function.cpp

```diff
@@ -0,0 +1,347 @@
+                                             const ProcessInfo& rProcessInfo)
+    {
+        KRATOS_TRY;
+
+        for(IndexType i = 0; i < mpNeighborConditions.size(); ++i)
+        {
+            Kratos::Condition& ng_cond_i = mpNeighborConditions[i];
+
+            if( rAdjointCondition.Id() == ng_cond_i.Id() )
+            {
+                Vector filter_vector;
+                ProcessInfo process_info = rProcessInfo;
+                this->ConstructFilterVector(rAdjointCondition, filter_vector, process_info);
+                noalias(rSensitivityGradient) = prod(rSensitivityMatrix, filter_vector);
+            }
+        }
+
+        KRATOS_CATCH("");
+    }
+
+
+    std::string AdjointNodalReactionResponseFunction::GetCorrespondingDisplacementDofLabel(std::string& rReactionDofLabel) const
+    {
+        KRATOS_TRY;
+
+        if (rReactionDofLabel == "REACTION_X")
+            return "DISPLACEMENT_X";
+        else if (rReactionDofLabel == "REACTION_Y")
+            return "DISPLACEMENT_Y";
+        else if (rReactionDofLabel == "REACTION_Z")
+            return "DISPLACEMENT_Z";
+        else if (rReactionDofLabel == "REACTION_MOMENT_X")
+            return "ROTATION_X";
+        else if (rReactionDofLabel == "REACTION_MOMENT_Y")
+            return "ROTATION_Y";
+        else if (rReactionDofLabel == "REACTION_MOMENT_Z")
+            return "ROTATION_Z";
+        else
+            KRATOS_ERROR << "Invalid reaction dof label!" << std::endl;
+
+        KRATOS_CATCH("");
+    }
+
+    void AdjointNodalReactionResponseFunction::PerformResponseVariablesCheck()
+    {
+        KRATOS_TRY;
+
+        // Check if variable for traced reaction is valid
+        if( !( KratosComponents< VariableComponent< VectorComponentAdaptor<array_1d<double, 3> > > >::Has(mTracedReactionLabel)) )
+            KRATOS_ERROR << "Specified traced DOF is not available. Specified DOF: " << mTracedReactionLabel << std::endl;
+        else
+        {
+            const VariableComponentType& r_traced_dof =
+                KratosComponents<VariableComponentType>::Get(mTracedReactionLabel);
+            KRATOS_ERROR_IF_NOT( mpTracedNode->SolutionStepsDataHas(r_traced_dof) )
+                << "Specified DOF is not available at traced node." << std::endl;
+        }
+
+        // Check if variable for traced dof is valid
+        if( !( KratosComponents< VariableComponent< VectorComponentAdaptor<array_1d<double, 3> > > >::Has(mTracedDisplacementDofLabel)) )
+            KRATOS_ERROR << "Specified traced DOF is not available. Specified DOF: " << mTracedDisplacementDofLabel << std::endl;
+        else
+        {
+            const VariableComponentType& r_traced_dof =
+                KratosComponents<VariableComponentType>::Get(mTracedDisplacementDofLabel);
+            KRATOS_ERROR_IF_NOT( mpTracedNode->SolutionStepsDataHas(r_traced_dof) )
+                << "Specified DOF is not available at traced node." << std::endl;
```

Comment:
All 4 checks use slightly different error messages but as far as i understand with the same information inside. Maybe this can be unified a bit.